web/src/components/core/PasswordInput.test.jsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { screen, fireEvent } from "@testing-library/react";
+import { plainRender } from "~/test-utils";
+import userEvent from "@testing-library/user-event";
+import PasswordInput from "./PasswordInput";
+import { _ } from "~/i18n";
+
+describe("PasswordInput Component", () => {
+  it("has initial state for password visibility as false", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+      />
```

Comment:
If finally `PF/FormGroups`s are moved back and props simplified as proposed, do not forget to clenaup the examples in the tests too. I.e., 

```suggestion
      <PasswordInput id="password" aria-abel={_("User password")} />
```